meson.build

```diff
@@ -0,0 +1,470 @@
+    dir_flatpak_intf = dep_flatpak.get_variable('interfaces_dir')
+endif
+
+dir_systemd_userunit = get_option('systemd-user-unit-dir')
+if dir_systemd_userunit == ''
+    # This is deliberately not ${libdir}: systemd units always go in
+    # .../lib, never .../lib64 or .../lib/x86_64-linux-gnu
+    dir_systemd_userunit = dir_prefix / 'lib' / 'systemd' / 'user'
+endif
+
+dir_dataroot = get_option('datarootdir')
+if dir_dataroot == ''
+    dir_dataroot = dir_datadir
+endif
+
+###### various include directories we'll use later
+# These are set here so meson handles the relative paths correctly,
+# makes life easier for us
+
+inc_builddir = include_directories('.')  # config.h
+inc_src = include_directories('src')
+
+###### plugins, dependencies, compiler setup
+
+i18n = import('i18n')
+gnome = import('gnome')
+pkgconfig = import('pkgconfig')
+
+cc = meson.get_compiler('c')
+cflags = [
+    '-Wno-unused-parameter',
+    '-Wno-sign-compare',
+    '-Wno-missing-field-initializers',
+]
+add_project_arguments(cc.get_supported_arguments(cflags), language : 'c')
+
+config_h = configuration_data()
+config_h.set('_GNU_SOURCE', 1)
+config_h.set_quoted('G_LOG_DOMAIN', 'xdg-desktop-portal')
+config_h.set_quoted('DATADIR', dir_datadir)
+config_h.set_quoted('LIBEXECDIR', dir_libexecdir)
+config_h.set_quoted('LOCALEDIR', dir_localedir)
+config_h.set_quoted('GETTEXT_PACKAGE', 'xdg-desktop-portal')
+config_h.set_quoted('PACKAGE_STRING', 'xdg-desktop-portal @0@'.format(meson.project_version()))
+if cc.has_function('renameat2')
+    config_h.set('HAVE_RENAMEAT2', 1)
+endif
+
+dep_glib = dependency('glib-2.0')
+dep_gio = dependency('gio-2.0')
+dep_gio_unix = dependency('gio-unix-2.0')
+dep_json_glib = dependency('json-glib-1.0')
+dep_fuse3 = dependency('fuse3', version: '>= 3.10.0')
+dep_geoclue = dependency('libgeoclue-2.0',
+                         version: '>= 2.5.2',
+                         required: get_option('geoclue'))
+dep_libportal = dependency('libportal',
+                           required: get_option('libportal'))
+dep_pipewire = dependency('libpipewire-0.3',
+                          version: '>= 0.2.90',
+                          required: get_option('pipewire'))
+
+prog_bwrap = find_program('bwrap')
+config_h.set_quoted('BWRAP', prog_bwrap.full_path())
+
+if dep_glib.version().version_compare('>= 2.60')
+    config_h.set('GLIB_VERSION_MIN_REQUIRED', 'GLIB_VERSION_2_60',
+                 description: 'Ignore massive GTimeVal deprecation warnings in 2.62')
+endif
+
+have_libportal = dep_libportal.found()
+if have_libportal
+    config_h.set('HAVE_LIBPORTAL', 1)
+endif
+
+have_geoclue = dep_geoclue.found()
+if have_geoclue
+    config_h.set('HAVE_GEOCLUE', 1)
+endif
+
+have_pipewire = dep_pipewire.found()
+if have_pipewire
+    config_h.set('HAVE_PIPEWIRE', 1)
+endif
+
+build_docbook = false
+prog_xmlto = find_program('xmlto', required: get_option('docbook-docs'))
+if not get_option('docbook-docs').disabled() and prog_xmlto.found()
```

Comment:
I don't understand the purpose of this conditional, the entire point of a feature option is so that if it is disabled, prog_xmlto will NOT be found -- it will log to the effect of "Program xmlto skipped: feature docbook docs is disabled".

Reply:
I'm thinking this may be more correct as:
```
if get_option('docbook-docs').auto() and prog_xmlto.found()
```
because it's for the case of `auto` where you may have xmlto but we still need to the fs check to check for the xml file to be present. unless there's some other way?

Reply:
No? You check if the file is present and then in the enabled() case you error out and in the auto case you leave `build_docbook = false`.

It is correct to skip only in the .disabled() case, and my point was that `prog_xmlto.found()` implicitly checks the disabled case *already*.

Reply:
No, my suggestion is correct. Delete the `not get_option('docbook-docs').disabled() and`

Just use:

```meson
if prog_xmlto.found()
```

Reply:
oh, indeed, thanks! changed locally, will be in the next push